### DISCUSSION TITLE
How to "fill in" missing edges

### DISCUSSION THREAD

**Discussion by agrigori-qc** [networkx/networkx#6746] [Q&A] [answered]
Hi!

Thanks for this excellent library :)

I am wondering how one could implicitly add missing edges? There is only a single path between any given pair of nodes. For example:

G.add_edge('a','b', r=1)
G.add_edge('b','c', r=2)

Then I would want the edge between 'a' and 'c' to be automatically added with the value r=3 (assuming a function can be provided, performing addition in this case)

**Comment by MridulS** [comment, marked as answer]
Maybe something like this should work for you?

``` python
import networkx as nx

def fill_in_empty_edges(G, weight):
    missing_edges = nx.complete_graph(list(G)).edges() - G.edges()
    for u, v in missing_edges:
        edge_w = nx.shortest_path_length(G, u, v, weight)
        G.add_edge(u, v, weight=edge_w)

G = nx.Graph()
G.add_edge('a','b', r=1)
G.add_edge('b','c', r=2)
fill_in_empty_edges(G, 'r')

```

**Reply by agrigori-qc** [reply]
@MridulS Thanks for the quick reply and good answer!

My usecase is slight more involved, I though I would simplify things in my question but ended up misleading, anyway:

Edges (**r**) would would not be just a number but represent a transformation matrix, and going between nodes would mean making this transformation. A better example:

R1 = numpy.random.rand(3,3)
R2 = numpy.random.rand(3,3)

G.add_edge('a','b', r=R1)
G.add_edge('b','c', r=R2)

# Adding missing edge
G.add_edge('a','c', r=R1 @ R2)

**Reply by rossbar** [reply]
I believe @MridulS 's suggestion addresses the original question sufficiently; i.e. using set operations to compare with edges in the corresponding complete graph. Edge attributes can of course be whatever you want for your application!
